implementations/elixir/ockam/ockam/lib/ockam/transport/tcp/handler.ex

```diff
@@ -35,6 +35,11 @@ defmodule Ockam.Transport.TCP.Handler do
     authorization = Keyword.get(handler_options, :authorization, [])
     tcp_wrapper = Keyword.get(handler_options, :tcp_wrapper, Ockam.Transport.TCP.DefaultWrapper)
 
+    # There are cases where we want to detect and close tcp connections after a long
+    # period without activity.  Mainly to handle the case of a server receiving connections
+    # from a long running client that "leak" them without closing.
+    idle_timeout = Keyword.get(handler_options, :idle_timeout, :infinity)
```

Comment:
I wonder if we should avoid `:infinity` and use a big timeout instead, like 30 days of inactivity.

Reply:
🤔  it might be a good idea indeed, but since we don't have explicit expectations about how transports (or channels) are used, I didn't want to change the existing behavior in any way unless configured to do so.    Apps using it can put these 30 days on their config.  (in most cases I bet 30 days won't even be reached as nodes are restarted, etc)